src/main/java/teammates/common/util/AppUrl.java

```diff
@@ -0,0 +1,49 @@
+package teammates.common.util;
+
+/**
+ * A specific implementation of {@link Url} used to encapsulate URLs of the application.
+ */
+public class AppUrl extends Url {
+    
+    private static final String BASE_WEB_URL_REGEX = "^(https?://[^/]+)";
+    
+    public AppUrl(String baseUrl, String url) {
+        super(baseUrl, makeRelativeUrl(url)); // force the stored URL to be relative
+        Assumption.assertTrue(baseUrl.matches(BASE_WEB_URL_REGEX + "$"));
+        String relativeUrl = makeRelativeUrl(url);
+        Assumption.assertTrue(relativeUrl.isEmpty() || relativeUrl.startsWith("/"));
```

Comment:
1. This problem might go away if we use the standard URL class, which is likely to be able to handle URLs of local resources.
2. I think this whole business of 'suppressing repetition' should not be there at all. The repetition is a result of the flawed API. If the caller has the relative path only, it should be like this: `Config.getAppUrl().setRelativePath(relativePath)`<br>
If the caller has the full URL, it should be `new AppUrl(fullUrl)`